include/nbl/builtin/hlsl/central_limit_blur/box_blur.hlsl

```diff
@@ -0,0 +1,297 @@
+			if( right <= last )
+			{
+				float32_t floorRight = prefixSumsAccessor.get( floor( right ) );
+				float32_t ceilRight = prefixSumsAccessor.get( ceil( right ) );
+				result = lerp( floorRight, ceilRight, frac( right ) );
+			}
+			else
+			{
+				switch( wrapMode )
+				{
+				case WRAP_MODE_CLAMP_TO_EDGE:
+				{
+					float32_t sumLast = prefixSumsAccessor.get( last );
+					float32_t sumLastMinusOne = prefixSumsAccessor.get( last - 1u );
+					result = ( right - float32_t( last ) ) * ( sumLast - sumLastMinusOne ) + sumLast;
+				} break;
+
+				case WRAP_MODE_CLAMP_TO_BORDER:
+				{
+					float32_t sumLast = prefixSumsAccessor.get( last );
+					result = sumLast + ( right - last ) * borderColor[ ch ];
+				} break;
+
+				case WRAP_MODE_REPEAT:
+				{
+					float32_t sumLast = prefixSumsAccessor.get( last );
+					float32_t sumModFloorRight = prefixSumsAccessor.get( fmod( floor( right ) - AXIS_DIM, AXIS_DIM ) );
+					float32_t sumModCeilRight = prefixSumsAccessor.get( fmod( ceil( right ) - AXIS_DIM, AXIS_DIM ) );
+					const float32_t v_floored = ceil( ( floor( right ) - last ) / AXIS_DIM ) * sumLast + sumModFloorRight;
+					const float32_t v_ceiled = ceil( ( ceil( right ) - last ) / AXIS_DIM ) * sumLast + sumModCeilRight;
+					result = lerp( v_floored, v_ceiled, frac( right ) );
+				} break;
+
+				case WRAP_MODE_MIRROR:
+				{
+					float32_t v_floored;
+					{
+						const uint32_t floored = uint32_t( floor( right ) );
+						const uint32_t d = floored - last; // distance from the right-most boundary, >=0
+
+						if( fmod( d, 2 * AXIS_DIM ) == AXIS_DIM )
+						{
+							v_floored = ( ( d + AXIS_DIM ) / AXIS_DIM ) * prefixSumsAccessor.get( last );
+						}
+						else
+						{
+							const uint32_t period = uint32_t( ceil( float32_t( d ) / AXIS_DIM ) );
+
+							if( ( period & 0x1u ) == 1 )
+								v_floored = period * prefixSumsAccessor.get( last ) + prefixSumsAccessor.get( last ) - prefixSumsAccessor.get( last - uint32_t( fmod( d, AXIS_DIM ) ) );
+							else
+								v_floored = period * prefixSumsAccessor.get( last ) + prefixSumsAccessor.get( fmod( d - 1, AXIS_DIM ) );
+						}
+					}
+
+					float32_t v_ceiled;
+					{
+						const uint32_t ceiled = uint32_t( ceil( right ) );
+						const uint32_t d = ceiled - last; // distance from the right-most boundary, >=0
+
+						if( fmod( d, 2 * AXIS_DIM ) == AXIS_DIM )
+						{
+							v_ceiled = ( ( d + AXIS_DIM ) / AXIS_DIM ) * prefixSumsAccessor.get( last );
+						}
+						else
+						{
+							const uint32_t period = uint32_t( ceil( float32_t( d ) / AXIS_DIM ) );
+
+							if( ( period & 0x1u ) == 1 )
+								v_ceiled = period * prefixSumsAccessor.get( last ) + prefixSumsAccessor.get( last ) - prefixSumsAccessor.get( last - uint32_t( fmod( d, AXIS_DIM ) ) );
+							else
+								v_ceiled = period * prefixSumsAccessor.get( last ) + prefixSumsAccessor.get( fmod( d - 1, AXIS_DIM ) );
+						}
+					}
+
+					result = lerp( v_floored, v_ceiled, frac( right ) );
+				} break;
```

Comment:
The old code necessitated using `MaxWorkItems` shared memory.

you can see here that it expects the whole prefix sum to be available in shared memory for arbitrary addressing

Reply:
Also the `AXIS_DIM` stuff is just weird here